MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+					self.fix_armature_target(
+						context, context.selected_objects, group)
+
+				# remove the empty added by making duplicates real.
+				if ob in bpy.data.objects[:] and ob.type == "EMPTY":
+					util.obj_unlink_remove(ob, True, context)
+				else:
+					for ob in context.selected_objects:
+						if ob.type != "EMPTY":
+							continue
+						elif ob.parent or ob.children:
+							continue
+						util.obj_unlink_remove(ob, True, context)
+				# TODO: Consider actually keeping the empty, and parenting the
+				# spawned items to this empty (not the defautl behavior)
+			else:
+				# change the draw size
+				if hasattr(ob, "empty_draw_size"):
+					ob.empty_draw_size = 0.25
+				else:
+					ob.empty_display_size = 0.25
+				# mark as active object
+				util.set_active_object(context, ob)
+
+			# Move source of group instance into excluded view layer
+			if util.bv28() and group:
+				move_assets_to_excluded_layer(context, [group])
+
+		self.track_param = self.block
+		return {'FINISHED'}
+
+	def fix_armature_target(self, context, new_objs, src_coll):
+		"""To address 2.8 bug where group made real might not update armature source"""
+
+		src_armas = [arma for arma in src_coll.objects
+			if arma.type == 'ARMATURE']
+		new_armas = [arma for arma in new_objs
+			if arma.type == 'ARMATURE']
+		src_armas_basnames = [util.nameGeneralize(obj.name) for obj in src_armas]
+		new_armas_basnames = [util.nameGeneralize(obj.name) for obj in new_armas]
+		if not new_armas:
+			return # nothing we can do to fix if new arma not existing
+		for obj in new_objs:
+			for mod in obj.modifiers:
+				# two scenarios:
+				# if 2.7: armature modifier already changed, copy over anim data
+				# if 2.8: swap modifier armature and also copy over anim data
+				if mod.type != 'ARMATURE':
+					continue
+				# if not (mod.object in new_objs or mod.object in src_armas):
+				#       continue # all good (from mod arma target perspective)
+				if mod.object not in src_armas and mod.object not in new_armas:
+					continue # unknown source, no way to know how to fix this
+				target_base = util.nameGeneralize(mod.object.name)
+				if target_base not in new_armas_basnames:
+					continue # shouldn't occur, but skip if it does
+
+				new_target = new_armas[new_armas_basnames.index(target_base)]
+				old_target = src_armas[src_armas_basnames.index(target_base)]
+
+				# for blender 2.8, bug is that target is still old_target
+				# but for blender 2.7, the mod target will already have changed
+				if old_target.animation_data:
+					new_target.animation_data_create()
+					new_target.animation_data.action = old_target.animation_data.action
+					conf.log("Updated animation of armature for instance of "+src_coll.name)
+
+				if mod.object in new_objs:
+					continue # was already the new object target for modifier
+				mod.object = new_target
+				conf.log("Updated target of armature for instance of "+src_coll.name)
+
+	def prep_collection(self, context, block, pre_groups):
```

Comment:
Optional refactor: ditto to above, move this into `spawn_util.py`, and reference back here for both entities and the meshswap file's equivalent usage.